### PR TITLE
Add missing -function-sections in Makefile in otherlibs

### PR DESCRIPTION
**Pull request by gretay-js**
The flag -function-sections was not passed to ocamlopt as needed in two of the Makefiles under otherlibs. These two Makefiles don't include the common Makefile which takes care of adding this option.

### CODE REVIEW

**Comment by mshinwell** [comment]
I suggest this is backported to 4.10.

**Comment by dra27** [comment]
The ultimate goal is to have only one common `Makefile` for `otherlibs`, but that’s a ton of work. However, rather than papering this over again, could this be done either by shifting these common definitions all the way up to `Makefile.common.in` (if that can work there - is this flag wanted everywhere ocamlopt is invoked?) or by splitting `Makefile.otherlibs.common` into parts which really are common to all three templates (macro definitions) and only for the “simple” libraries (rules for unix, str, etc.)?

**Comment by gretay-js** [comment]
@dra27 I'm sorry for leaving your comment unanswered for so long (not sure how I missed it).

As you suggested, I moved the definition of OPTCOMPFLAGS to `Makefile.common.in`. It is the only place where FUNCTION_SECTIONS variable is checked and the flag `-function-sections` is added to OPTCOMPFLAGS. So far, these changes preserve the behavior we had before. 

In addition, the last commit also moves the check for FLAMBDA to `Makefile.common.in` in order to add  "-O3" to OPTCOMPFLAGS.  It preserves the behavior of OPTCOMPFLAGS in `stdlib` and `otherlib`. It changes OPTCOMPFLAGS in the "main" Makefile to use "-O3" option when configured with `FLAMBDA`. The tests pass, it doesn't seem to take longer, and it is nice and uniform. Is there a particular reason it shouldn't be compiled with -O3?

**Review by dra27 — APPROVED** [review]
Thanks for the changes, and sorry for my lag approving!

File: Makefile
Comment (dra27) [inline Makefile]:
```suggestion
```

File: Makefile.common.in
Comment (dra27) [inline Makefile.common.in]:
```suggestion
```

File: stdlib/Makefile
Comment (dra27) [inline stdlib/Makefile]:
```suggestion
```

**Review by dra27 — COMMENTED** [review]
Given that it still short a Changes entry, 3 nits of extra blank lines which I don't think are needed!

**Comment by dra27** [comment]
Good to go once it has a `Changes` entry and I agree with @mshinwell that this should go to 4.10 as well.

**Comment by dra27** [comment]
@Octachron - if this is going in 4.10, does that the mean the `Changes` entry on `trunk` should also be the "4.10 Maintenance Branch" bit, or in the "Working Version"(/4.11) bit?

**Comment by dra27** [comment]
Thanks for the updates @gretay-js - I'll merge and cherry-pick this once CI is green!

**Comment by gretay-js** [comment]
Thank you!

**Comment by dra27** [comment]
Ping @Octachron, just to confirm which `Changes` section this can go in - I'll fix that after merge

**Comment by Octachron** [comment]
Sorry, I missed your earlier question. Since we are planning to have a 4.10.1 release, before the 4.11.0 release, I think it is fine to have the entry in the future 4.10.1 section.

**Comment by dra27** [comment]
Thanks!

**Pull request merged**
